From: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Date: Sat, 4 Jul 2020 03:04:38 +0300
Subject: dt/dtoverlays: ov5647: Add parameter to set camera module rotation

Add a rotation DT overlay parameter to allow specifying the camera
module mounting rotation.

Signed-off-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
---
 arch/arm/boot/dts/overlays/README             | 5 +++--
 arch/arm/boot/dts/overlays/ov5647-overlay.dts | 6 ++++++
 2 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/overlays/README b/arch/arm/boot/dts/overlays/README
index eb561ef272e0..651acea3194b 100644
--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -1809,8 +1809,9 @@ Name:   ov5647
 Info:   Omnivision OV5647 camera module.
         Uses Unicam 1, which is the standard camera connector on most Pi
         variants.
-Load:   dtoverlay=ov5647
-Params: <None>
+Load:   dtoverlay=ov5647,<param>=<val>
+Params: rotation                Mounting rotation of the camera sensor (0 or
+                                180, default 0)
 
 
 Name:   ov7251
diff --git a/arch/arm/boot/dts/overlays/ov5647-overlay.dts b/arch/arm/boot/dts/overlays/ov5647-overlay.dts
index 5eba3cb9fb91..fd1e7a457f69 100644
--- a/arch/arm/boot/dts/overlays/ov5647-overlay.dts
+++ b/arch/arm/boot/dts/overlays/ov5647-overlay.dts
@@ -21,6 +21,8 @@
 				pwdn-gpios = <&gpio 41 1>, <&gpio 32 1>;
 				clocks = <&ov5647_clk>;
 
+				rotation = <0>;
+
 				port {
 					ov5647_0: endpoint {
 						remote-endpoint = <&csi1_ep>;
@@ -83,4 +85,8 @@
 			};
 		};
 	};
+
+	__overrides__ {
+		rotation = <&ov5647>,"rotation:0";
+	};
 };
